spring-integration-gcp/src/main/java/org/springframework/integration/gcp/storage/GCSSession.java

```diff
@@ -0,0 +1,192 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.integration.gcp.storage;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.google.cloud.WriteChannel;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.integration.file.remote.session.Session;
+import org.springframework.util.Assert;
+import org.springframework.util.StreamUtils;
+
+/**
+ * @author João André Martins
+ */
+public class GCSSession implements Session<BlobInfo> {
+
+	private Storage gcs;
+
+	private static final Log LOGGER = LogFactory.getLog(GCSSession.class);
+
+	public GCSSession(Storage gcs) {
+		this.gcs = gcs;
+	}
+
+	@Override
+	public boolean remove(String path) throws IOException {
+		String[] tokens = getBucketAndObjectFromPath(path);
+
+		return tokens.length < 3
+				? this.gcs.delete(tokens[1])
+				: this.gcs.delete(tokens[1], tokens[2]);
```

Comment:
Something wrong is here. You do `path.split("/", 2);`, so, you can't get more than 2 tokens and the second one will be like `folder/blob`.
I might miss something, so elaborate, please.

Reply:
The way I understand it is `remove()` can be used to delete either GCS blobs or buckets.
You're right that the `2` in `path.split("/", 2)` needs to be a `3` so that, for `/bucket/path/blob`, the result is `["", "bucket", "path/blob"]`.

Reply:
???
If you would like to remove just bucket, you have to use just bucket name and we will have `tokens.length == 1`. If you would like to remove a blob, the path should be with bucket and blob over a `/` . Therefore `tokens.length == 2`, but never `3`.

I don't understand everything rest you try to explain to me, but I would like to see a test case for this remove, for both `bucket` and `bucket/path/blob` use-cases.

Thanks

Reply:
Sorry for the confusion. I thought the paths in there would be `/bucket/file`, but there is no starting `/`, so 3 doesn't make sense.